app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -227,143 +206,125 @@ export const snapKeyringBuilder = (
           });
         };
 
-        const learnMoreLink =
-          'https://support.metamask.io/managing-my-wallet/accounts-and-addresses/how-to-add-accounts-in-your-wallet/';
-
-        // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
-        const skipConfirmation =
-          isSnapPreinstalled(snapId) && !displayConfirmation;
-        // If confirmation dialog are skipped, we consider the account creation to be confirmed until the account name dialog is closed
-        const accountCreationConfirmationResult =
-          skipConfirmation ||
-          (await showAccountCreationDialog(snapId, controllerMessenger));
-
-        if (!accountCreationConfirmationResult) {
-          // User has cancelled account creation
-          await handleUserInput(accountCreationConfirmationResult);
-
-          throw new Error('User denied account creation');
-        }
-
-        // We need to temporarily add the account to the keyring to show the account name suggestion dialog
         try {
-          await handleUserInput(accountCreationConfirmationResult);
-          await persistKeyringHelper();
-        } catch (e) {
-          // Error occurred while creating the account
-          const error = (e as Error).message;
-
-          await showError(
-            controllerMessenger,
-            snapId,
-            {
-              icon: IconName.UserCircleAdd,
-              title: t('snapAccountCreationFailed'),
-            },
-            {
-              message: t(
-                'snapAccountCreationFailedDescription',
-                snapName,
-              ) as string,
-              learnMoreLink,
-              error,
-            },
-          );
-
-          trackSnapAccountEvent(MetaMetricsEventName.AccountAddFailed);
-
-          throw new Error(
-            `Error occurred while creating snap account: ${error}`,
-          );
-        }
-
-        const accountNameConfirmationResult =
-          await showAccountNameSuggestionDialog(
-            snapId,
-            address,
-            controllerMessenger,
-            accountNameSuggestion,
-          );
-
-        if (accountNameConfirmationResult) {
-          try {
-            setSelectedAccountHelper(address);
+          const learnMoreLink =
+            'https://support.metamask.io/managing-my-wallet/accounts-and-addresses/how-to-add-accounts-in-your-wallet/';
+
+          // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
+          const skipConfirmation =
+            isSnapPreinstalled(snapId) && !displayConfirmation;
+          // If confirmation dialog are skipped, we consider the account creation to be confirmed until the account name dialog is closed
+          const accountCreationConfirmationResult =
+            skipConfirmation ||
+            (await showAccountCreationDialog(snapId, controllerMessenger));
```

Comment:
Why are we not calling and end for this flow like we are for the `addAccountApprovalId` [here](https://github.com/MetaMask/metamask-extension/blob/feat/define-account-name-during-creation-2/app/scripts/lib/snap-keyring/snap-keyring.ts#L325)?

Reply:
For the `addAccount` we only want to use 1 approval flow because it will allow us to reuse the same popup for both requests. This fixes the flaky test issue where the driver is trying to switch to the second popup.